hibernate-core/src/main/java/org/hibernate/generator/values/GeneratedValueBasicResultBuilder.java

```diff
@@ -103,19 +103,13 @@ public BasicValuedModelPart getModelPart() {
 	}
 
 	private static int columnIndex(JdbcValuesMetadata jdbcResultsMetadata, BasicValuedModelPart modelPart) {
-		try {
+		if ( jdbcResultsMetadata.getColumnCount() == 1 ) {
+			return 0;
```

Comment:
Just to be safe maybe?

```suggestion
			assert modelPart.isEntityIdentifierMapping() || jdbcResultsMetadata.resolveColumnPosition(
					getActualGeneratedModelPart( modelPart ).getSelectionExpression()
			) == 1;
			return 0;
```

Reply:
Well why would we need to assert this? If the result set only contains one column it will always be on index `0`, independently if it is an identifier or not.

Reply:
I'm thinking, maybe the lookup would cause an exception that we would otherwise not see.